waku/v2/protocol/filter/options.go

```diff
@@ -53,19 +54,41 @@ type (
 		wg             *sync.WaitGroup
 	}
 
-	FilterParameters struct {
+	FilterFullNodeParameters struct {
 		Timeout        time.Duration
 		MaxSubscribers int
 		pm             *peermanager.PeerManager
+		limitR         rate.Limit
+		limitB         int
 	}
 
-	Option func(*FilterParameters)
+	FullNodeOption func(*FilterFullNodeParameters)
```

Comment:
I used FullNodeOption because both lightnode and fullnode have their own options, also, using `Filter` stutters when you use the package name due to repeating the word "filter", i.e. "filter.FilterFullNodeOption` vs `filter.FullNodeOption`.

Hm. I should change the parameters struct name too.